versioned_docs/version-v1.3/user-manuals/sidecarset.md

```diff
@@ -0,0 +1,523 @@
+
+Currently known cases that utilize the SidecarSet hot upgrade mechanism:
+- [ALIYUN ASM](https://help.aliyun.com/document_detail/193804.html) implements lossless upgrade of Data Plane in Service Mesh.
+
+### Inject Pod Metadata (Annotations)
+**FEATURE STATE:** Kruise v1.3.0
+
+SidecarSet support inject pod annotations, as follows:
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: SidecarSet
+spec:
+  containers:
+    ...
+  patchPodMetadata:
+  - annotations:
+      oom-score: '{"log-agent": 1}'
+      custom.example.com/sidecar-configuration: '{"command": "/home/admin/bin/start.sh", "log-level": "3"}'
+    patchPolicy: MergePatchJson
+  - annotations:
+      apps.kruise.io/container-launch-priority: Ordered
+    patchPolicy: Overwrite | Retain
+```
+patchPolicy is the injected policy, as follows:
+  - **Retain:** By default, if annotation[key]=value exists in the Pod, the original value of the Pod will be retained. Inject annotations[key]=value2 only if annotation[key] does not exist in the Pod.
+  - **Overwrite:** Corresponding to Retain, when annotation[key]=value exists in the Pod, it will be overwritten value2.
+  - **MergePatchJson:** Corresponding to Overwrite, the annotations value is a json string. If the annotations[key] does not exist in the Pod, it will be injected directly. If it exists, do a json value merge.
+For example: annotations[oom-score]='{"main": 2}' exists in the Pod, after injection, the value json is merged into annotations[oom-score]='{"log-agent": 1, "main": 2}'.
+
+**Note:** When the patchPolicy is Overwrite and MergePatchJson, the annotations can be updated synchronously when the SidecarSet in-place update the Sidecar Container.
+However, if only the annotations are modified, it will not take effect. It must be in-place update together with the sidecar container image.
+When patchPolicy is Retain, the annotations will not be updated when the SidecarSet in-place update the Sidecar Container.
+
+According to the above configuration, when the sidecarSet is injected into the sidecar container, it will inject Pod annotations synchronously, as follows:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    apps.kruise.io/container-launch-priority: Ordered
+    oom-score: '{"log-agent": 1, "main": 2}'
+    custom.example.com/sidecar-configuration: '{"command": "/home/admin/bin/start.sh", "log-level": "3"}'
+name: test-pod
+spec:
+  containers:
+    ...
+```
+
+#### Permission Control
+SidecarSet should not modify any configuration outside the sidecar container from permission perspective. Metadata, as an important configuration of Pod, should not be modified by sidecarSet by default.
+
+Objectively, sidecar does need to have some annotations or labels injected into the pod as well. In order to meet the needs of sidecar and security considerations.
```

Comment:
> SidecarSet should not modify any configuration outside the sidecar container from permission perspective

sidecarSet should not modify any configuration outside the sidecar container for security consideration

> Objectively, sidecar does need to have some annotations or labels injected into the pod as well. In order to meet the needs of sidecar and security considerations.

just delete this sentence since it provide no useful information